### PR TITLE
feat(aws-applicationautoscaling): Allow autoscaling with "M out of N" datapoints

### LINKED ISSUE #17433: (aws-applicationautoscaling): Support datapointsToAlarm in StepScalingPolicy
### Description

`StepScalingPolicy` currently supports setting `evaluationPeriods` for the upper and lower autoscaling alarms, but not `datapointsToAlarm`. This means that to trigger the alarm, all data points in the evaluation period must trigger.

### Use Case

I would like to create an autoscaling alarm that triggers when only a portion of the datapoints in the evaluation period have triggered.

### Proposed Solution

Add an optional property to `StepScalingPolicy` for datapointsToAlarm that is passed to the underlying `Alarm`.

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by stephenwiebe**
This PR closes #17433. It adds a `datapointsToAlarm` property to the `StepScalingPolicy` construct which allows auto-scaling activities to trigger when only a portion of the data points in the evaluation periods are breaching.

Motivation: Some metrics may have a certain amount of noise/randomness and in these cases it may make more sense to not require that all data points must be breaching for auto-scaling activity to trigger.
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by stephenwiebe** [comment]
@comcalvi Any way this could be merged in? There's not much to this change and it'd be nice to have this functionality.

File: packages/@aws-cdk/aws-applicationautoscaling/lib/step-scaling-policy.ts [hunk 1]
```diff
@@ -130,6 +146,7 @@ export class StepScalingPolicy extends CoreConstruct {
         alarmDescription: 'Lower threshold scaling alarm',
         comparisonOperator: cloudwatch.ComparisonOperator.LESS_THAN_OR_EQUAL_TO_THRESHOLD,
         evaluationPeriods: props.evaluationPeriods ?? 1,
+        datapointsToAlarm: props.datapointsToAlarm,
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-applicationautoscaling/lib/step-scaling-policy.ts:149]: CloudFormation docs state that this is an integer with minimum value 1. We should validate this prop before we pass it in here, and throw an error if validation fails.

File: packages/@aws-cdk/aws-applicationautoscaling/lib/step-scaling-policy.ts [hunk 2]
```diff
@@ -160,6 +177,7 @@ export class StepScalingPolicy extends CoreConstruct {
         alarmDescription: 'Upper threshold scaling alarm',
         comparisonOperator: cloudwatch.ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
         evaluationPeriods: props.evaluationPeriods ?? 1,
+        datapointsToAlarm: props.datapointsToAlarm,
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-applicationautoscaling/lib/step-scaling-policy.ts:180]: CloudFormation docs state that this is an integer with minimum value 1. We should validate this prop before we pass it in here, and throw an error if validation fails.

File: packages/@aws-cdk/aws-applicationautoscaling/test/step-scaling-policy.test.ts [hunk 3]
```diff
@@ -227,6 +227,43 @@ describe('step scaling policy', () => {
 
 
   });
+
+  test('step scaling with evaluation period & data points to alarm configured', () => {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-applicationautoscaling/test/step-scaling-policy.test.ts:231]: Once the `dataPointsToAlarm` verification is added, can we add a test to verify that the error is correctly thrown?

**Review by comcalvi — COMMENTED** [review]
Looks great! A few minor changes and this will be ready to be merged.

**Comment by stephenwiebe** [comment]
@comcalvi Ok, I think this should be resolved now.

**Review by comcalvi — APPROVED** [review]
LGTM!

**Review by comcalvi — CHANGES_REQUESTED** [review]
To fix the linter error, could you add the new property and brief of explanation of it to the README?

**Review by comcalvi — APPROVED** [review]
LGTM! Thanks for adding this to the readme 😄

**Comment by stephenwiebe** [comment]
@comcalvi What's the next step to get it merged in? I wasn't sure if I needed to merge the master branch back into mine, so I did that but it looks like it might have nullified your review.

**Review by comcalvi — APPROVED** [review]

**Pull request merged**